.cloudbuild/prombench.yaml

```diff
@@ -0,0 +1,335 @@
+              export PROMETHEUS_IMAGE_VERSION="$${DEFAULT_COLLECTOR_TAG}"
+            else
+              echo "$${COLLECTOR_TAG_OR_CANCEL}" > /workspace/PROMETHEUS_IMAGE_VERSION
+              export PROMETHEUS_IMAGE_VERSION="$${COLLECTOR_TAG_OR_CANCEL}"
+            fi
+          fi
+          
+          echo "The release/base GMP Collector's tag is set to: $${PROMETHEUS_IMAGE_VERSION}."
+
+        fi
+
+  - id: "Checking cluster's existance"
+    name: gcr.io/cloud-builders/gcloud
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        if $(gcloud container clusters describe "$${CLUSTER_NAME}" --project="$${GKE_PROJECT_ID}" --location="$${ZONE}" >/dev/null 2>&1)
+        then
+          CREATE_CLUSTER="false"
+          echo "$${CREATE_CLUSTER}" > /workspace/CREATE_CLUSTER
+          echo "The cluster (name: $${CLUSTER_NAME}) already exists. Therefore, skipping cluster creation."
+        else
+          CREATE_CLUSTER="true"
+          echo "$${CREATE_CLUSTER}" > /workspace/CREATE_CLUSTER
+          echo "The cluster (name: $${CLUSTER_NAME}) does not exist, the cluster will get created before creating the nodes and triggering the benchmark process."
+        fi
+
+  - id: "Setting gcloud credentials for triggering the benchmark"
+    name: gcr.io/cloud-builders/gcloud
+    secretEnv: ['ADC']
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        echo "$${ADC}" > /workspace/adc.json
+
+  - id: "Triggering Benchmark Action"
+    name: gcr.io/google.com/cloudsdktool/cloud-sdk
+    secretEnv: ['GH_PAT']
+    env:
+    - '_GIT_REPO=$_GIT_REPO' # GoogleCloudPlatform/prometheus
+    - '_PR_NUMBER=$_PR_NUMBER'
+    entrypoint: "bash"
+    args:
+      - "-c"
+      - |
+        # This function posts comments to GitHub PRs
+        function post_to_github() {
+          URL="$1"
+          BODY="$2"
+
+          curl --silent -L -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $${GH_PAT}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "$${URL}" \
+            -d '{"body": "'"$${BODY}"'"}'
+        }
+
+        # Some of the environment variables (e.g. CREATE_CLUSTER) are overwritten in previous steps and the
+        # new values need to be used in this step. "export"ing the variables does not help and hence the new
+        # values are written to a local file and the corresponding file is read in this step.
+        # Reference: https://cloud.google.com/build/docs/configuring-builds/pass-data-between-steps
+        function get_or_set_env_var() {
+          ENV_VAR_NAME="$1"
+          if [ -f "/workspace/$${ENV_VAR_NAME}" ]
+          then
+            export "$${ENV_VAR_NAME}=$(cat /workspace/$${ENV_VAR_NAME})"
+          fi
+        }
+
+        # All the environment variables created or updated in any of the previous steps must be read
+        # from the local file and exported in this step.
+        get_or_set_env_var "CANCEL_BENCHMARK";
```

Comment:
> can create each file Setting environment variables step to avoid the check on line 204

This will make the whole thing consistent, which is good! But, then if later on the environment variables are added or deleted, the file creation step/function would need to be updated as well. Contributors might forget to update the file creation step/function as there are only 3 variables that get updated (as of now). Rest all the env vars are used as is.